server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -57,16 +75,64 @@ object Email {
   type Size = Long Refined NonNegative
   val Zero: Size = 0L
 
-  def sanitizeSize(value: Long): Size = {
+  private[mail] def sanitizeSize(value: Long): Size = {
     val size: Either[String, Size] = refineV[NonNegative](value)
     size.fold(e => {
       logger.error(s"Encountered an invalid Email size: $e")
       Zero
     },
       refinedValue => refinedValue)
   }
+
+  private[mail] def parseAsMime4JMessage(firstMessage: MessageResult): Try[Message] = {
+    val defaultMessageBuilder = new DefaultMessageBuilder
+    defaultMessageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE)
+    defaultMessageBuilder.setDecodeMonitor(DecodeMonitor.SILENT)
+    val inputStream = firstMessage.getFullContent.getInputStream
+    val resultMessage = Try(defaultMessageBuilder.parseMessage(inputStream))
+    inputStream.close()
```

Comment:
it would not work on something else than memory input stream, as it will close the stream before it will be used.
Because the Try.apply parameter is call by name (lazy)

Maybe you can do something like : 

Try(resultMessage.fold(close stream on error and re-throw error, close stream on success))